src/main/java/eu/eutampieri/catacombs/ui/World.java

```diff
@@ -48,12 +42,16 @@ public World(final TileMap tileMap, final DungeonGame game) {
 
         final SingleObjectFactory objectFactory = new SingleObjectFactoryImpl(this.tileMap);
         final Random rand = new Random();
-        this.entities.addAll(objectFactory.spawnSome(700, (x, y, tm) -> {
+        this.entities.addAll(objectFactory.spawnSome(5, (x, y, tm) -> {
             final int healingPower = rand.nextInt(101);
             return new SimplePotion(healingPower, "Potion", x, y);
         }));
 
         this.player = (Player) mf.spawnSome(1, (x, y, tm) -> new Player(x, y, "", tm)).get(0);
+        /* for boss debugging purpose */
+        this.entities.addAll(mf.spawnAt((this.player.getPosX() - this.camera.getXOffset()) / AssetManagerProxy.getMapTileSize() + 1,
+                (this.player.getPosY() - this.camera.getYOffset()) / AssetManagerProxy.getMapTileSize() + 1,
+                Boss::new).stream().map((x) -> (GameObject) x).collect(Collectors.toList()));
```

Comment:
Now that we know that it works and renders correctly, we may spawn it randomly

Reply:
yessir